src/qt3utils/applications/qt3ple/main.py

```diff
@@ -0,0 +1,503 @@
+                                                                       model.get_start,
+                                                                       model.get_end + model.step_size])
+            cbar = self.fig.colorbar(artist, ax=ax)
+
+            #if self.log_data is False:
+            #    cbar.formatter.set_powerlimits((0, 3))
+
+            self.ax.set_xlabel('Pixels')
+            self.ax.set_ylabel('Voltage (V)')
+
+    def update_plot(self, model, grid) -> None:
+
+        for ii, reader in enumerate(model.readers):
+            y_data = model.outputs[model.current_frame-1][reader]
+            x_control = model.scanned_control[model.current_frame-1]
+            ax = self.fig.add_subplot(grid[1, ii])
+            ax.plot(x_control, y_data, color='k', linewidth=1.5)
+
+    def reset(self) -> None:
+        self.ax.cla()
+
+    def set_onclick_callback(self, f) -> None:
+        self.onclick_callback = f
+
+    def onclick(self, event) -> None:
+        pass
+
+
+class SidePanel():
+    def __init__(self, root, scan_range) -> None:
+        frame = tk.Frame(root.root)
+        frame.pack(side=tk.LEFT, fill=tk.BOTH, expand=True)
+
+        row = 0
+        tk.Label(frame, text="Scan Settings", font='Helvetica 16').grid(row=row, column=0, pady=10)
+        row += 1
+        self.start_button = tk.Button(frame, text="Start Scan")
+        self.start_button.grid(row=row, column=0)
+        self.stop_button = tk.Button(frame, text="Stop Scan")
+        self.stop_button.grid(row=row, column=1)
+        self.save_scan_button = tk.Button(frame, text="Save Scan")
+        self.save_scan_button.grid(row=row, column=2)
+        row += 1
+        tk.Label(frame, text="Voltage Range (V)").grid(row=row, column=0)
+        self.voltage_start_entry = tk.Entry(frame, width=10)
+        self.voltage_end_entry = tk.Entry(frame, width=10)
+        self.voltage_start_entry.insert(10, scan_range[0])
+        self.voltage_end_entry.insert(10, scan_range[1])
+        self.voltage_start_entry.grid(row=row, column=1)
+        self.voltage_end_entry.grid(row=row, column=2)
+
+        row += 1
+        tk.Label(frame, text="Number of Pixels").grid(row=row, column=0)
+        self.num_pixels = tk.Entry(frame, width=10)
+        self.num_pixels.insert(10, 150)
+        self.num_pixels.grid(row=row, column=1)
+
+        row += 1
+        tk.Label(frame, text="Number of Scans").grid(row=row, column=0)
+        self.scan_num_entry = tk.Entry(frame, width=10)
+        self.scan_num_entry.insert(10, 10)
+        self.scan_num_entry.grid(row=row, column=1)
+
+        row += 1
+        tk.Label(frame, text="Sweep Time").grid(row=row, column=0)
+        self.sweep_time_entry = tk.Entry(frame, width=10)
+        self.sweep_time_entry.insert(10, 3)
+        self.sweep_time_entry.grid(row=row, column=1)
+
+        row += 1
+        self.do_downsweep = tk.IntVar(value=0)
+        tk.Label(frame, text="Downsweep Time").grid(row=row, column=0)
```

Comment:
maybe instead of down sweep it's better to say backward sweep? It is more generic.